internal/corazawaf/rulegroup.go

```diff
@@ -29,8 +29,13 @@ func (rg *RuleGroup) Add(rule *Rule) error {
 		return nil
 	}
 
-	if rule.ID_ != 0 && rg.FindByID(rule.ID_) != nil {
-		return fmt.Errorf("there is a another rule with id %d", rule.ID_)
+	// rule id is mandatory and must be unique ID for all SecRule/SecAction.
+	if rule.SecMark_ == "" { // means its SecRule/SecAction
+		if rule.ID_ == 0 {
+			return fmt.Errorf("rule id is missing, rule residing in file %s at line %d", rule.File_, rule.Line_)
+		} else if rg.FindByID(rule.ID_) != nil {
+			return fmt.Errorf("there is a another rule with id %d", rule.ID_)
+		}
```

Comment:
nit: If the previous if condition matches, the code returns, we can avoid the nested `else` condition
```suggestion
		}
		if rg.FindByID(rule.ID_) != nil {
			return fmt.Errorf("duplicated rule id %d", rule.ID_)
		}
```

Reply:
I still prefer if...else if when conditions are logically related. It is more readable (I feel).
Anyway, I have addressed your suggestion.